mlflow_client/backends.py

```diff
@@ -134,11 +140,51 @@ def end_run(self):
         pass
 
 
-class LocalBackend(Backend):
+class WithHooks(object):
+    """
+    Class decorator
+    Decorates a Backend to provide hooks on some methods
+    """
+    def __init__(self):
+        pass
+
+    def __call__(self, cls):
+        class BackendWithHooks(cls):
+            """
+            SubClass of the decorated Backend that implements Hooks
+            """
+            def __init__(self, *args, **kwargs):
```

Comment:
Ok but add docstring for `hooks_uri`.

Reply:
That's what I mentionned before, by using a class decorator to add default behaviour, the 'hooks_uri' parameter cannot be documented by autocompletion when trying to do LocalBackend() for example

Even if I had docstring to the subclass